src/esm_runscripts/filedicts.py

```diff
@@ -63,10 +63,13 @@ def foo(self, *args, **kwargs):
     @functools.wraps(method)
     def inner_method(self, *args, **kwargs):
         if self.allowed_to_be_missing:
-            logger.warning(
-                f"Skipping {method.__qualname__} as this file ({self}) is allowed to be missing!"
-            )
-            return None  # None is the default return, but let us be explicit here, as it is a bit confusing
+            try:
+                return method(self, *args, **kwargs)
+            except (FileNotFoundError, IOError):
```

Comment:
Just thought about one thing here, what about also adding `SystemExit` to cover up for `user_erros`?